src/lightning_lite/strategies/fairscale.py

```diff
@@ -120,3 +121,56 @@ def _reinit_optimizers_with_oss(self, optimizers: List[Optimizer]) -> List["OSS"
                 optimizers[x] = zero_optimizer
                 del optimizer
         return optimizers
+
+
+class DDPSpawnShardedStrategy(DDPSpawnStrategy):
+    """Optimizer sharded training provided by FairScale."""
+
+    def setup_module_and_optimizers(
+        self, module: Module, optimizers: List[Optimizer]
+    ) -> Tuple[Module, List[Optimizer]]:
+        """Wraps the model and optimizers with fairscale components.
+
+        Return:
+            The model wrapped into a :class:`~fairscale.nn.data_parallel.ShardedDataParallel` module
+            and a list of optimizer wrapped in :class:~`fairscale.optim.OSS`.
+        """
+        optimizers = self._reinit_optimizers_with_oss(optimizers)
+        model = ShardedDataParallel(module, sharded_optimizer=optimizers, **self._ddp_kwargs)
+        return model, optimizers
+
+    @contextmanager
+    def block_backward_sync(self, module: Module) -> Generator:
+        """Blocks syncing gradients behaviour on backwards pass.
+
+        This is useful for skipping sync when accumulating gradients, reducing communication overhead
+        Returns: context manager with sync behaviour off
+        """
+        if isinstance(module, ShardedDataParallel):
+            with module.no_sync():
+                yield None
+        else:
+            yield None
+
+    @classmethod
+    def register_strategies(cls, strategy_registry: Dict) -> None:
+        strategy_registry.register(
+            "ddp_sharded_spawn_find_unused_parameters_false",
+            cls,
+            description="DDP Spawn Sharded Strategy with `find_unused_parameters` as False",
+            find_unused_parameters=False,
+        )
+        strategy_registry.register(
+            "ddp_sharded_spawn",
+            cls,
+            description=f"{cls.__class__.__name__}",
+        )
+
+    def _reinit_optimizers_with_oss(self, optimizers: List[Optimizer]) -> List["OSS"]:
```

Comment:
This implementation should exactly match the one in `DDSharded`, no? Did they accidentally diverge in PL?

Reply:
Yes, they seem to differ only in the multi-node aspect. While ddp spawn technically does support multi-node, it is not as convenient when launching processes, and hence I guess the original author(s) thought the regular ddp implementation should be the one with the multi-node optimizations. 

I'll think about what to do here.

Reply:
Made them identical in [185dd64](https://github.com/Lightning-AI/lightning/pull/14675/commits/185dd6451744ef96b2cd1e34cf2bc162d08cf782)

Awaiting further reduction by #14707